File: chatbot-service/test_shared_vectors.py
"""
교차검증: shared_test_vectors.json에 대해 Python mask_pii가 기대한 대로 판정하는지 확인.
같은 파일을 was/test-shared-vectors.js가 JS 쪽에서도 검증한다 - 두 결과가 어긋나면
Python/JS 마스킹 규칙이 드리프트됐다는 신호.
"""
import json
import sys
import unittest
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))
from pii_masking import mask_pii

VECTORS_PATH = Path(__file__).resolve().parent.parent / "shared_test_vectors.json"


class TestSharedVectors(unittest.TestCase):
    def test_all_vectors(self):
        with open(VECTORS_PATH, encoding="utf-8") as f:
            vectors = json.load(f)

        for v in vectors:
            with self.subTest(text=v["text"]):
                result = mask_pii(v["text"])
                has_url = "[MASKED_INTERNAL_URL]" in result
                has_key = "[MASKED_API_KEY]" in result
                self.assertEqual(has_url, v["expect_internal_url"], f"internal_url 판정 불일치: {v['text']!r} -> {result!r}")
                self.assertEqual(has_key, v["expect_api_key"], f"api_key 판정 불일치: {v['text']!r} -> {result!r}")


if __name__ == "__main__":
    unittest.main(verbosity=2)

File: was/test-shared-vectors.js
// 교차검증: shared_test_vectors.json에 대해 JS maskPii가 Python과 같은 판정을 내리는지 확인.
const fs = require("fs");
const path = require("path");
const assert = require("assert");
const { maskPii } = require("./crypto-utils");

const vectorsPath = path.join(__dirname, "..", "shared_test_vectors.json");
const vectors = JSON.parse(fs.readFileSync(vectorsPath, "utf-8"));

let failed = 0;
for (const v of vectors) {
  const result = maskPii(v.text);
  const hasUrl = result.includes("[MASKED_INTERNAL_URL]");
  const hasKey = result.includes("[MASKED_API_KEY]");
  try {
    assert.strictEqual(hasUrl, v.expect_internal_url, `internal_url 판정 불일치: "${v.text}" -> "${result}"`);
    assert.strictEqual(hasKey, v.expect_api_key, `api_key 판정 불일치: "${v.text}" -> "${result}"`);
    console.log(`ok   - ${v.text}`);
  } catch (e) {
    console.log(`FAIL - ${v.text}`);
    console.log(`       ${e.message}`);
    failed += 1;
  }
}

console.log(`\n${failed === 0 ? "모두 통과 (Python과 판정 일치)" : failed + "개 불일치"}`);
process.exit(failed === 0 ? 0 : 1);
